stores/profileStore/controls.js

```diff
@@ -1,5 +1,7 @@
-export async function REQUEST_PROFILE() {
-  const response = await fetch('/Data/users.json');
-  const results = await response.json();
-  return results;
+export async function REQUEST_PROFILE(action) {
+  const { userName } = action;
+  if (userName) {
+    const response = await fetch(`/api/${userName}`);
```

Comment:
That's strange, but I see now why you do this. Note that the API endpoint doesn't need to do anything with the URL of where it's used.

If the page has the URL `/username` the API can be `api/abbas/getusers`. They're not connected at all. 

This URL should be `api/users?userId=${userName}`.